lib/trino-filesystem-azure/src/main/java/io/trino/filesystem/azure/AzureFileSystem.java

```diff
@@ -465,9 +465,7 @@ private boolean isHierarchicalNamespaceEnabled(AzureLocation location)
 
```

Comment:
> Some operations, like checking if directory exists, could fail if paths are not consistently encoded and when they contain special characters.

Can we please add a test to `AbstractTestAzureFileSystem` or `AbstractTestTrinoFileSystem` for these use cases as well?

Reply:
I added one test case in `AbstractTestTrinoFileSystem.testRename`, I hope that's enough. I ensured it was failing before applying changes in `AzureFilesystem`.